modules/dists/OldReplicatedDist.chpl

```diff
@@ -0,0 +1,652 @@
+
+Note that the above behavior may change in the future. In particular,
+we are considering changing it so that replication is never observable.
+For example, only the local replicand would be accessed in all cases.
+
+
+**Example**
+
+  .. code-block:: chapel
+
+    const Dbase = {1..5};  // a default-distributed domain
+    const Drepl: domain(1) dmapped ReplicatedDist() = Dbase;
+    var Abase: [Dbase] int;
+    var Arepl: [Drepl] int;
+
+    // only the current locale's replicand is accessed
+    Arepl[3] = 4;
+
+    // these iterate over Dbase;
+    // only the current locale's replicand is accessed
+    forall (b,r) in zip(Abase,Arepl) do b = r;
+    Abase = Arepl;
+
+    // these iterate over Drepl; each replicand of Drepl
+    // will be zippered against (and copied from) the entire Abase
+    forall (r,b) in zip(Arepl,Abase) do r = b;
+    Arepl = Abase;
+
+    // sequential zippering will detect difference in sizes
+    // (if multiple locales)
+    for (b,r) in zip(Abase,Arepl) ... // error
+    for (r,b) in zip(Arepl,Abase) ... // error
+
+
+**Constructor Arguments**
+
+The ``ReplicatedDist`` class constructor is defined as follows:
+
+  .. code-block:: chapel
+
+    proc ReplicatedDist(
+      targetLocales: [] locale = Locales,
+      purposeMessage: string = "used to create a ReplicatedDist")
+
+The array ``targetLocales`` must be "consistent", as defined below.
+
+The optional ``purposeMessage`` may be useful for debugging
+when the constructor encounters an error.
+
+
+**Features/Limitations**
+
+* Only rectangular domains are presently supported.
+
+* Serial iteration over a replicated domain (or array) visits the indices
+  (or array elements) of all replicands *from the current locale*.
+
+* When replicating over user-provided array of locales, that array
+  must be "consistent" (see below).
+
+"Consistent" array requirement:
+
+* The array of desired locales, if passed explicitly as ``targetLocales``
+  to the ReplicatedDist constructor, must be "consistent".
+
+* The array ``A`` is "consistent" if
+  for each ``ix`` in ``A.domain``, this holds: ``A[ix].id == ix``.
+
+* Tip: if the domain of your ``targetLocales`` cannot be described
+  as a rectangular domain (whether strided, multi-dimensional,
+  and/or sparse), make the domain associative over the `int` type.
+
+*/
+class ReplicatedDist : BaseDist {
+  var targetLocDom : domain(Locales.idxType);
+
+  // the desired locales (an array of locales)
+  const targetLocales : [targetLocDom] locale;
+}
+
+
+// constructor: replicate over the given locales
+// (by default, over all locales)
+proc ReplicatedDist.ReplicatedDist(targetLocales: [] locale = Locales,
+                                   purposeMessage: string = "used to create a ReplicatedDist",
+                                   param warning = true)
+{
+  if warning then
+    compilerWarning("The `ReplicatedDist` domain map is deprecated.  Please switch over to using `Replicated` instead.");
```

Comment:
I thought about this, but opted not to for the sake of brevity.  Since we don't suggest that this is merely a renaming (i.e., we are telling them to use a different domain map), I feel like we're covered.  Are you OK with that?